var/spack/repos/builtin/packages/py-cutadapt/package.py

```diff
@@ -15,23 +15,30 @@ class PyCutadapt(PythonPackage):
     pypi = "cutadapt/cutadapt-1.13.tar.gz"
     git = "https://github.com/marcelm/cutadapt.git"
 
+    version("4.1", sha256="be745ff24adfb4a3eaf715dfad0e2ccdfad7792ef00c1122adf4fbf3aed9227b")
     version("2.10", sha256="936b88374b5b393a954852a0fe317a85b798dd4faf5ec52cf3ef4f3c062c242a")
     version("2.9", sha256="cad8875b461ca09cea498b4f0e78b0d3dcd7ea84d27d51dac4ed45080bf1499e")
     version("2.5", sha256="ced79e49b93e922e579d0bb9d21298dcb2d7b7b1ea721feed484277e08b1660b")
     version("1.13", sha256="aa9f2c1f33dc081fe94f42b1250e4382b8fb42cabbf6e70a76ff079f211d5fc0")
 
-    depends_on("python@2.6:", type=("build", "run"), when="@1.13")
-    depends_on("python@3.4:", type=("build", "run"), when="@2.0:2.5")
-    depends_on("python@3.5:", type=("build", "run"), when="@2.6:")
+    # universal deps
     depends_on("py-setuptools", type=("build", "run"))
     depends_on("py-setuptools-scm", type="build", when="@2.0:")
+    # 4.1 deps
+    depends_on("python@3.7:", type=("build", "run"), when="@4.1:")
+    depends_on("py-dnaio@0.7.1:", type=("build", "run"), when="@4.1:")
+    depends_on("py-xopen@1.1:", type=("build", "run"), when="@4.1:")
+    # older version deps
+    depends_on("python@2.6:", type=("build", "run"), when="@1.13")
+    depends_on("python@3.4:", type=("build", "run"), when="@2.0:2.5")
+    depends_on("python@3.5:", type=("build", "run"), when="@2.6:2.10")
```

Comment:
Spack no longer supports Python 3.6 and older, so you don't technically have to specify these at all, but you can leave them in if you want.

Reply:
I was actually looking for your feed back for this, I didn't see either of these packages in the big eol pr. Would you like me to just remove the older versions entirely?

Reply:
I've not tested, but looking at the setup.py and the like for the old version it sounds like 3.7 would probably work without issue for everything except possibly 1.13

Reply:
Most packages are forwards compatible, so there's no reason to remove the old versions. But there's also no reason to say that certain versions support Python 2.6+ and others support Python 3.4+ when Spack only supports Python 3.7+ anyway.